Doc: Update documentation for trace output (#3139)
carbon-language/carbon-lang: explorer/README.md

Please refer to
[Fuzzer documentation](https://github.com/carbon-language/carbon-lang/blob/trunk/explorer/fuzzing/README.md).

## Explorer's Trace Output

Explorer's Trace Output refers to a detailed record of program phases and their

> internal processes a program goes through when executed using the `explorer`, it
> also records things like changes in memory and action stack that describes the

Reviewer: ```suggestion
internal processes a program goes through when executed using the `explorer`. It
also records things like changes in memory and action stack that describes the
```

state of the program.

> Tracing can be turned on (using the `--trace_file=...` option or `.verbose`
> target).

Reviewer: ```suggestion
Tracing can be turned on using the `--trace_file=...` option. Explorer tests
can be run with tracing enabled by using the `<testname>.verbose` test target.
```

I think it's better to list the user-facing trace support first, with the unit tests left as more of an afterthought.

By default, `explorer` prints the state of the program and each step that is
performed during execution for the file containing the main function when

> tracing is enabled, tracing for different phases and file contexts can be

Reviewer: ```suggestion
tracing is enabled. Tracing for different phases and file contexts can be
```

-   `include`: Includes trace output for all.
-   By default, tracing is only enabled for the `main` file context.

**Note (for developers):** Two
[RAII](https://en.cppreference.com/w/cpp/language/raii) classes

> `SetProgramPhase` and `SetFileContext`, are there for setting program phase and

Reviewer: ```suggestion
`SetProgramPhase` and `SetFileContext` are provided for setting program phase and
```
